platform/android/src/jni.cpp

```diff
@@ -1515,10 +1515,12 @@ void setOfflineRegionObserver(JNIEnv *env, jni::jobject* offlineRegion_, jni::jo
             }
 
             // Error object
-            jni::jobject* jerror = &jni::NewObject(*env2, *offlineRegionErrorClass, *offlineRegionErrorConstructorId);
```

Comment:
Did you try using `jni::PushLocalFrame` here? I'm curious if that alone would have fixed the issue.